src/google/create-token.ts

```diff
@@ -0,0 +1,51 @@
+import fs = require('fs');
+import readline = require('readline');
+import { Auth } from 'googleapis';
+
+// If modifying these scopes, delete token.json. Create a fresh one. At the moment, we only use the Google Docs API, hence the scope is for documents only.
+const SCOPES = ['https://www.googleapis.com/auth/documents'];
+const TOKEN_PATH = process.env.TOKEN_PATH || 'token.json';
+
+// Scopes with the calendar included will become:
+// const SCOPES = ['https://www.googleapis.com/auth/documents', 'https://www.googleapis.com/auth/calendar.readonly'];
+
+/**
+ * Get and store new token after prompting for user authorization, and then
+ * execute the given callback with the authorized OAuth2 client.
+ *
+ * @remark The file token.json stores the user's access and refresh tokens, and is
+ * created automatically when the authorization flow completes for the first time.
+ *
+ * @param {google.auth.OAuth2} oAuth2Client The OAuth2 client to get token for.
+ */
+export async function getNewToken(auth: Auth.OAuth2Client) {
+	try {
+		const authUrl = auth.generateAuthUrl({
+			access_type: 'offline',
+			scope: SCOPES,
+		});
+		console.log(
+			'This script will create a token.json file for you to authenticate Google APIs \n',
+		);
+		console.log('Authorize this app by visiting this url:', authUrl);
+		const rl = readline.createInterface({
+			input: process.stdin,
+			output: process.stdout,
+		});
+		rl.question('Enter the code from that page here: ', (code: string) => {
+			rl.close();
+			auth.getToken(code, (err, token) => {
+				if (err || !token) {
+					return console.error('Error retrieving access token', err, token);
+				}
+				fs.writeFileSync(TOKEN_PATH, JSON.stringify(token));
+				auth.setCredentials(token);
+			});
+		});
+	} catch (err) {
+		console.log('Cannot create token.json file: ', err);
+	}
+}
+
+// If someone needs to just generate a token.json file, they can uncomment the next line and run this file as per the instruction in the README.md file to create token.json file.
+// getNewToken();
```

Comment:
@Hades32 @somombo @20k-ultra I have JS/TS doubt about this. If I uncomment this file above and run the project using `npm start`. This file gets executed for some reason during runtime and the console logs show it asking for token. Any reason why that happens? 

Could it be that on transpilation from TS to JS the file gets executed or on compile-time when files are being parsed? This module gets picked up and executed. 

The main purpose I had for this file was: 
1. Creating token files through the same code. DRY. 
2. Creating token files using a standalone script, because creating the token files in the middle of a running meetbot server was a very messy implementation which I tried to fix here. 

I can't uncomment the line above, because it starts executing the entire thing even when no one called it.

Reply:
@vipulgupta2048 that's a top-level function call. So it's completely expected, that that function gets called as soon as that file gets loaded - probably because it's imported somewhere.

If you want to share logic between interactive and non-interactive code, just keep the module without auto-execution and have a separate file importing that module doing the interactive part/the method call. The latter file would never be imported but could be run explicitely e.g. by an npm task

Reply:
@Hades32 Oh I see, great idea. Implemented.